libs/pipelines/top_counties_pipeline.py

```diff
@@ -0,0 +1,71 @@
+from collections import namedtuple
+import logging
+
+from libs.enums import Intervention
+from libs import validate_results
+from libs import build_dod_dataset
+from libs import dataset_deployer
+from api.can_predictions import CANPredictionAPIRow, CANPredictionAPI, _Projections, _HospitalBeds
+from libs.datasets import results_schema as rc
+
+logger = logging.getLogger(__name__)
+PROD_BUCKET = "data.covidactnow.org"
+
+TopCountiesPipelineResult = namedtuple(
+    "TopCountiesPipelineResult", ["key", "projection_df"]
+)
+
+
+def run_projections(
+    input_file, intervention: Intervention, run_validation=True
+) -> TopCountiesPipelineResult:
+    """Run county level projections for a specific intervention.
+
+    Args:
+        input_file: Input file to load model output results from.
+        intervention: Intervention type to summarize.
+        run_validation: If true runs validation on generated shapefiles
+            and dataframes.
+
+    Returns: TopCountiesPipelineResult objects for county data.
+    """
+    # Run County level projections
+    counties_key_name = f"counties.{intervention.name}"
+    counties_df = build_dod_dataset.get_usa_by_county_with_projection_df(
+        input_file, intervention.value
+    )
+    if run_validation:
+        validate_results.validate_counties_df(counties_key_name, counties_df)
+
+    county_results = TopCountiesPipelineResult(
+        counties_key_name, counties_df
+    )
+
+    return county_results
+
+def generate_api(intervention_result: TopCountiesPipelineResult): 
+    projection = intervention_result.projection_df
+
+    api_results = []
+    # TODO: I think pandas doesn't really vibe with iterating, but it's 3000 and the other option is applying 
+    # things in a weird way so idk this is anti pandas but w/e 
```

Comment:
oh that works, or you can do `projection.to_dict(orient='records')` and it will return a list of dictionaries. 
that might be the same thing that's happening here? what type is `county_row`?

But I actually think iterating here is the right thing to do in general.

Reply:
okay i will take out this comment